enumerable_method.rb

```diff
@@ -123,38 +151,64 @@ def my_map
     return to_enum(:my_map) unless block_given?
 
     newarray = []
+    if self.class == Array
+      my_each do |x|
+        newarray << if yield x
+                      true
+                    else
+                      false
+                    end
+      end
+    else
+      i = first
+      while i <= last
+        newarray << yield(i) unless yield(i).nil?
+        i += 1
+      end
+    end
+    newarray
+  end
+
+  def operate_inject_for_simbol(acum, newarray, param)
+    newarray.my_each do |item|
+      acum = operation_simbol(param, acum, item)
+    end
+    acum
+  end
+
+  def new_auto_array(first, last)
+    autoarray = []
     i = first
     while i <= last
-      newarray << yield(i) unless yield(i).nil?
+      autoarray << i
       i += 1
     end
-    newarray
+    autoarray
   end
 
-  def my_inject(param = nil)
-    if self.class == Array
+  def my_inject(param = nil, _default_param = nil)
+    if param.class == Symbol
+      return nil if length.zero?
+
+      acum = operate_inject_for_simbol(self[0], self[1..length], param)
+    elsif self.class == Array
       acum = self[0]
       newarray = self[1..length]
       newarray.my_each do |item|
         acum = yield(item, acum)
       end
     else
-      autoarray = []
-      i = first
-      while i <= last
-        autoarray << i
-        i += 1
-      end
+      autoarray = new_auto_array(first, last)
       acum = autoarray[0]
       temparray = autoarray[1..autoarray.length]
       temparray.my_each do |item|
         acum = yield(item, acum)
       end
     end
-    acum = if_nil_acum_mul(param, acum)
+    acum
   end
```

Comment:
`my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)).
- While I tried to execute the following codes, it did not provide me the expected result. Remeber when we provide sybol as parameter we dont have to provide a block.
```ruby
[1, 2 , 3].my_inject(4) { |prod, n| prod * n }  #should provide me 24 but got 6
(1...4).my_inject(:+) #should return 6
(1...4).my_inject(1, :+)  #should return 7
```

Reply:
I solved the issues